lesson14/js/main.js

```diff
@@ -0,0 +1,174 @@
+	const loadMenuBtn = document.getElementById('js-loadmenuBtn');
+	loadMenuBtn.addEventListener(
+		'click',
+		(event) => {
+			const inputNumber = document.getElementById('number');
+			if (!inputNumber.value === false) {
+				getInputValue();
+				callMenuContents();
+			} else {
+				alert('未入力です。');
+			}
+		},
+		false
+	);
+}
+
+function closeModalWindow() {
+	const closeModlBtn = document.getElementById('js-closeModal');
+
+	closeModlBtn.addEventListener(
+		'click',
+		() => {
+			const modalContainer = document.getElementById('js-modalContainer');
+			const modalWindow = document.createElement('div');
+			const bgFixed = document.getElementById('js-bgFixed');
+			removeInnerHTML(modalWindow);
+			removeInnerHTML(menuLists);
+			document.body.removeAttribute('style');
+			bgFixed.remove();
+			modalContainer.remove();
+		},
+		false
+	);
+}
+
+function getInputValue() {
+	const inputNumber = document.getElementById('number');
+	console.log(inputNumber.value);
+}
+
+//CREATE MENU contents
+function showLoadingImg() {
+	const menuContainer = document.getElementById('js-menu_container');
+	const createLoadingBox = document.createElement('div');
+	createLoadingBox.classList.add('ly_loading');
+	createLoadingBox.id = 'js-loading';
+	const loadingImg = document.createElement('img');
+	loadingImg.src = 'images/loading-circle.gif';
+	loadingImg.classList.add('el_loadingImg');
+	menuContainer.appendChild(createLoadingBox).appendChild(loadingImg);
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function removeInnerHTML(element) {
+	element.innerHTML = '';
+}
+
+//menulist
+function createMenuList(data) {
+	const menuContainer = document.getElementById('js-menu_container');
+	const fragment = document.createDocumentFragment();
+	removeInnerHTML(menuLists);
+
+	data.forEach((value) => {
+		const listItem = document.createElement('li');
+		const listImg = document.createElement('img');
+		listImg.src = value.img;
+		listImg.alt = value.alt;
+		listImg.width = 40;
+		listImg.height = 40;
+
+		const listAnchor = document.createElement('a');
+		listAnchor.href = value.a;
+		listAnchor.textContent = value.text;
+		fragment.appendChild(listItem).appendChild(listAnchor).appendChild(listImg);
+	});
+	menuContainer.appendChild(menuLists).appendChild(fragment);
+}
+
+const MENU_DATA_URL = 'https://jsondata.okiba.me/v1/json/Aticp210816214718';
+async function fetchMenuContents() {
+	try {
+		const response = await fetch(MENU_DATA_URL);
+		const json = await response.json();
+		return json.data;
+	} catch {
+		console.error('Could not get the value');
+	}
+}
+
+async function callMenuContents() {
+	showLoadingImg();
+	const menuContainer = document.getElementById('js-menu_container');
+	try {
+		const menuContents = await fetchMenuContents();
+		if (menuContents.length > 0) {
+			createMenuList(menuContents);
+		} else {
+			menuContainer.textContent = '表示するメニューがありませんでした';
+		}
```

Comment:
[IMO]
I thought it would be better not to conditional branch in the try to avoid getting another error during the try.

Reply:
I think that if may be used in try depending on the work done in try.

In this case, we are getting the menu data,
This is because the value is not always included.

If there is no value, I wanted to make an error, so I am using an if statement here.